lineax/_solver/blocktridiagonal.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any
+from typing_extensions import TypeAlias
+
+import jax
+import jax.lax as lax
+import jax.numpy as jnp
+from jaxtyping import Array, PyTree
+
+from .._operator import (
+    AbstractLinearOperator,
+    blocktridiagonal,
+    is_blocktridiagonal,
+    MatrixLinearOperator,
+)
+from .._solution import RESULTS
+from .._solve import AbstractLinearSolver, linear_solve
+from .misc import (
+    pack_structures,
+    PackedStructures,
+    ravel_vector,
+    transpose_packed_structures,
+    unravel_solution,
+)
+
+
+_BlockTridiagonalState: TypeAlias = tuple[tuple[Array, Array, Array], PackedStructures]
+
+
+class BlockTridiagonal(AbstractLinearSolver[_BlockTridiagonalState]):
+    """Block tridiagonal solver for linear systems, using the Thomas algorithm."""
+
+    def init(self, operator: AbstractLinearOperator, options: dict[str, Any]):
```

Comment:
For most of the other dense matrix solvers, `Solver.init` also performs the necessary factorizations (see https://github.com/patrick-kidger/lineax/blob/main/lineax/_solver/lu.py for an example)

This could also be done here by splitting up the compute method a bit, I have a version like this that I've been using here:
https://gist.github.com/f0uriest/c4764bd8f7882a4b63457b1585b41ad9

Generally the factorization costs n^3, but the backsolve only n^2 so it's useful if you're re-using the operator against multiple RHS in a way that can't be easily vmapped (ie, using a block tridiagonal preconditioner for one of the iterative solvers)

(my version also has a `reverse` option for the factorization which I've found is needed in some cases when spectrally discretizing PDEs. this can easily be done by the user as a pre-processing step so maybe not needed but might be a useful option)